controller/utils.go

```diff
@@ -13,6 +16,28 @@ import (
 	longhorn "github.com/longhorn/longhorn-manager/k8s/pkg/apis/longhorn/v1beta2"
 )
 
+const (
+	podRecreateInitBackoff = 1 * time.Second
+	podRecreateMaxBackoff  = 120 * time.Second
+	backoffGCPeriod        = 12 * time.Hour
+)
+
+// newBackoff returns a flowcontrol.Backoff and starts a background GC loop.
+func newBackoff() *flowcontrol.Backoff {
+	backoff := flowcontrol.NewBackOff(podRecreateInitBackoff, podRecreateMaxBackoff)
+
+	// Start a background GC loop.
+	go func() {
+		ticker := time.NewTicker(backoffGCPeriod)
+		defer ticker.Stop()
+		for range ticker.C {
+			backoff.GC()
+		}
+	}()
+
+	return backoff
+}
```

Comment:
NIT: It might be worth mentioning the potential risk if this function is ever reused in a different context later. 
```go
// newBackoff returns a flowcontrol.Backoff and starts a background GC loop.
//
// This function starts a background goroutine that cannot be gracefully stopped,
// making it suitable only for processes that run for the entire application's
// lifetime. Using it in short-lived or frequently called contexts will lead to
// goroutine leaks. For such cases, refactor to accept a context.Context for proper
// cleanup.
```

Reply:
Or refactor it to:
```go
func newBackoff(ctx context.Context) *flowcontrol.Backoff {
	backoff := flowcontrol.NewBackOff(podRecreateInitBackoff, podRecreateMaxBackoff)

	go func() {
		ticker := time.NewTicker(backoffGCPeriod)
		defer ticker.Stop()
		for {
			select {
			case <-ctx.Done():
				return
			case <-ticker.C:
				backoff.GC()
			}
		}
	}()

	return backoff
}
```

Reply:
The PR is merged by Mergify.
@bachmanity1 Could you help create a PR to address @c3y1huang's comment? Thank you.

Reply:
@bachmanity1 
For now, I think adding a comment should be sufficient. Since Mergify already merged :sweat_smile: , I suggest opening another PR for improvement.

cc @derekbit